### ISSUE TITLE
Switch from LiteDB to SQLLite for local storage

### ISSUE THREAD

**Issue by Frooxius** [Yellow-Dog-Man/Resonite-Issues#2616]
### Is your feature request related to a problem? Please describe.

We've been using LiteDB for local data storage for a while. However years of my accumulated experience with this library, I've ran into lots of stability issues, data corruption and other problems, where I feel like I need to treat the library with silk gloves, otherwise it might explode.

The stability and reliability of this library isn't as good as I'd like it to be and I'd like us to transition away from it at some point.

### Describe the solution you'd like

SQLLite is probably the most popular solution for local storage out there and it's known for its solid reliability and very heavy set of unit tests to ensure that.

This seems like the best candidate to switch to.

https://www.sqlite.org/

### Describe alternatives you've considered

We could look into other libraries, but there's also inherent risk of it not being a safe bet as SQLLite is.

### Additional Context

I've recently tried to update LiteDB from 5.0.17 to 5.0.21 and it made everything explode - lots of exceptions when reading the data from existing database file. 

This is unacceptable kind of behavior for something that stores user data - especially with minor version upgrade.

**Comment by Banane9** [comment]
SQLite is love <3

**Comment by Dessix** [comment]
I'd like to recommend [libsql](https://github.com/tursodatabase/libsql) for consideration as a potential alternative- it's a SQLite fork which drops some of the prior project goals and accepts community contributions.

**Comment by Frooxius** [comment]
@Dessix Thanks for the recommendation, but we really need something solid and battle tested and using a fork like that comes with risks, which we don't want to undertake again, since we got badly burnt with LiteDB already.


I think we'll need to prioritize this soon(ish), because the issues keep accumulating and they're hard or straight up impossible to work around

**Comment by shininghero** [comment]
It might also be a curious exercise to try using MariaDB for the local storage.  It's licensed under GNU GPLv2, and is surprisingly used by several components of KDE's desktop environment.

**Comment by Gawdl3y** [comment]
SQLite is much more applicable to our use-case since it's an in-process database.

SQLite is incredibly fast and resilient, even with multiple readers and/or writers on the same system. It kind of wrecks every client-server database (such as MariaDB) in performance. It's not very scalable since it's specifically a local, in-process database, but that's perfect for the needs of the Resonite clients.

**Comment by shininghero** [comment]
Of course, anything that provides reliable function _and_ minimizes future tech-debt.  Not sure why KDE went with MariaDB anyways.  That's something to mull over in the off-topic Discord channels anyways.
